// @lipana/sdk/dist/Lipana.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Lipana = void 0;
const Transactions_1 = require("./resources/Transactions");
const PaymentLinks_1 = require("./resources/PaymentLinks");
const Webhooks_1 = require("./resources/Webhooks");
const ApiKeys_1 = require("./resources/ApiKeys");
const Payouts_1 = require("./resources/Payouts");
class Lipana {
    constructor(config) {
        if (!config.apiKey) {
            throw new Error('Lipana API key is required');
        }
        this.apiKey = config.apiKey;
        this.environment = config.environment || 'production';
        this.baseUrl = config.baseUrl || this.getDefaultBaseUrl();
        // Initialize resources
        const sharedConfig = {
            apiKey: this.apiKey,
            baseUrl: this.baseUrl,
            environment: this.environment,
        };
        this.transactions = new Transactions_1.Transactions(sharedConfig);
        this.paymentLinks = new PaymentLinks_1.PaymentLinks(sharedConfig);
        this.webhooks = new Webhooks_1.Webhooks(sharedConfig);
        this.apiKeys = new ApiKeys_1.ApiKeys(sharedConfig);
        this.payouts = new Payouts_1.Payouts(sharedConfig);
    }
    getDefaultBaseUrl() {
        return this.environment === 'sandbox'
            ? 'https://api-sandbox.lipana.dev/v1'
            : 'https://api.lipana.dev/v1';
    }
    /**
     * Get the current SDK version
     */
    static get VERSION() {
        return '1.0.2';
    }
}
exports.Lipana = Lipana;
