// @lipana/sdk/dist/errors/LipanaError.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LipanaError = void 0;
class LipanaError extends Error {
    constructor(message, statusCode, code, errors) {
        super(message);
        this.name = 'LipanaError';
        this.statusCode = statusCode;
        this.code = code;
        this.errors = errors;
        // Maintains proper stack trace for where our error was thrown
        if (Error.captureStackTrace) {
            Error.captureStackTrace(this, LipanaError);
        }
    }
    /**
     * Check if error is due to authentication failure
     */
    isAuthenticationError() {
        return this.statusCode === 401 || this.code === 'AUTHENTICATION_FAILED';
    }
    /**
     * Check if error is due to validation failure
     */
    isValidationError() {
        return this.statusCode === 400 || this.code === 'VALIDATION_ERROR';
    }
    /**
     * Check if error is due to rate limiting
     */
    isRateLimitError() {
        return this.statusCode === 429 || this.code === 'RATE_LIMIT_EXCEEDED';
    }
    /**
     * Check if error is a server error
     */
    isServerError() {
        return this.statusCode !== undefined && this.statusCode >= 500;
    }
}
exports.LipanaError = LipanaError;
