// @lipana/sdk/dist/utils/http.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpClient = void 0;
const axios_1 = __importDefault(require("axios"));
const LipanaError_1 = require("../errors/LipanaError");
class HttpClient {
    constructor(config) {
        this.apiKey = config.apiKey;
        this.client = axios_1.default.create({
            baseURL: config.baseUrl,
            timeout: 30000,
            headers: {
                'Content-Type': 'application/json',
                'User-Agent': `Lipana-Node-SDK/1.0.2`,
            },
        });
        // Request interceptor to add API key
        this.client.interceptors.request.use((config) => {
            config.headers['x-api-key'] = this.apiKey;
            return config;
        }, (error) => {
            return Promise.reject(error);
        });
        // Response interceptor for error handling
        this.client.interceptors.response.use((response) => response, (error) => {
            if (error.response) {
                const { status, data } = error.response;
                const message = data?.message || error.message || 'An error occurred';
                const code = data?.code;
                const errors = data?.errors;
                throw new LipanaError_1.LipanaError(message, status, code, errors);
            }
            else if (error.request) {
                throw new LipanaError_1.LipanaError('No response received from server', undefined, 'NETWORK_ERROR');
            }
            else {
                throw new LipanaError_1.LipanaError(error.message);
            }
        });
    }
    async get(url, config) {
        const response = await this.client.get(url, config);
        return response.data?.data || response.data;
    }
    async post(url, data, config) {
        const response = await this.client.post(url, data, config);
        return response.data?.data || response.data;
    }
    async put(url, data, config) {
        const response = await this.client.put(url, data, config);
        return response.data?.data || response.data;
    }
    async delete(url, config) {
        const response = await this.client.delete(url, config);
        return response.data?.data || response.data;
    }
}
exports.HttpClient = HttpClient;
