// @lipana/sdk/dist/resources/PaymentLinks.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentLinks = void 0;
const http_1 = require("../utils/http");
class PaymentLinks {
    constructor(config) {
        this.http = new http_1.HttpClient(config);
    }
    /**
     * List all payment links
     */
    async list(params) {
        const queryParams = new URLSearchParams();
        if (params?.page)
            queryParams.append('page', params.page.toString());
        if (params?.limit)
            queryParams.append('limit', params.limit.toString());
        if (params?.status)
            queryParams.append('status', params.status);
        const query = queryParams.toString();
        return this.http.get(`/payment-links${query ? `?${query}` : ''}`);
    }
    /**
     * Retrieve a specific payment link
     */
    async retrieve(id) {
        return this.http.get(`/payment-links/${id}`);
    }
    /**
     * Create a new payment link
     */
    async create(params) {
        return this.http.post('/payment-links', params);
    }
    /**
     * Update a payment link
     */
    async update(id, params) {
        return this.http.put(`/payment-links/${id}`, params);
    }
    /**
     * Delete a payment link
     */
    async delete(id) {
        return this.http.delete(`/payment-links/${id}`);
    }
    /**
     * Get public payment link (no auth required)
     */
    async getPublic(slug) {
        return this.http.get(`/payment-links/public/${slug}`);
    }
    /**
     * Initiate payment on a payment link
     */
    async initiatePayment(slug, phone, amount) {
        return this.http.post(`/payment-links/public/${slug}/pay`, { phone, amount });
    }
}
exports.PaymentLinks = PaymentLinks;
