// @lipana/sdk/dist/resources/Payouts.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Payouts = void 0;
const axios_1 = __importDefault(require("axios"));
const LipanaError_1 = require("../errors/LipanaError");
class Payouts {
    constructor(config) {
        this.apiKey = config.apiKey;
        this.baseUrl = config.baseUrl;
        this.environment = config.environment;
        this.client = axios_1.default.create({
            baseURL: this.baseUrl,
            headers: {
                'Authorization': `Bearer ${this.apiKey}`,
                'Content-Type': 'application/json'
            }
        });
        // Add interceptor for error handling
        this.client.interceptors.response.use(response => response, error => {
            if (error.response) {
                throw new LipanaError_1.LipanaError(error.response.data.message || 'An error occurred', error.response.status, error.response.data.code, error.response.data.errors);
            }
            throw new LipanaError_1.LipanaError(error.message, 500);
        });
    }
    /**
     * List payouts
     */
    async list(params) {
        const response = await this.client.get('/payouts', { params });
        return response.data;
    }
    /**
     * Retrieve a payout
     */
    async retrieve(id) {
        const response = await this.client.get(`/payouts/${id}`);
        return response.data.data;
    }
    /**
     * Request a manual payout (withdraw funds)
     */
    async request(params) {
        const response = await this.client.post('/payouts/request', params);
        return response.data.data;
    }
    /**
     * Send money to a specific recipient
     */
    async sendToRecipient(recipientId, amount) {
        const response = await this.client.post(`/payouts/recipient/${recipientId}`, { amount });
        return response.data.data;
    }
    /**
     * Send money to all active recipients
     */
    async sendToAllRecipients(amount) {
        const response = await this.client.post('/payouts/recipients/bulk', { amount });
        return response.data.data;
    }
    /**
     * Top up account via MPESA STK Push
     */
    async topUp(params) {
        const response = await this.client.post('/payouts/topup', params);
        return response.data.data;
    }
}
exports.Payouts = Payouts;
