// @lipana/sdk/dist/resources/Transactions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transactions = void 0;
const http_1 = require("../utils/http");
class Transactions {
    constructor(config) {
        this.http = new http_1.HttpClient(config);
    }
    /**
     * List all transactions
     */
    async list(params) {
        const queryParams = new URLSearchParams();
        if (params?.page)
            queryParams.append('page', params.page.toString());
        if (params?.limit)
            queryParams.append('limit', params.limit.toString());
        if (params?.status)
            queryParams.append('status', params.status);
        if (params?.startDate)
            queryParams.append('startDate', params.startDate);
        if (params?.endDate)
            queryParams.append('endDate', params.endDate);
        const query = queryParams.toString();
        return this.http.get(`/transactions${query ? `?${query}` : ''}`);
    }
    /**
     * Retrieve a specific transaction
     */
    async retrieve(id) {
        return this.http.get(`/transactions/${id}`);
    }
    /**
     * Create a new transaction
     */
    async create(params) {
        return this.http.post('/transactions', params);
    }
    /**
     * Initiate an STK push payment
     */
    async initiateStkPush(params) {
        return this.http.post('/transactions/push-stk', params);
    }
    /**
     * Update transaction status
     */
    async updateStatus(id, status) {
        return this.http.put(`/transactions/${id}/status`, { status });
    }
}
exports.Transactions = Transactions;
