// @lipana/sdk/dist/resources/Webhooks.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhooks = void 0;
const http_1 = require("../utils/http");
const crypto = __importStar(require("crypto"));
class Webhooks {
    constructor(config) {
        this.http = new http_1.HttpClient(config);
    }
    /**
     * Get webhook settings
     */
    async getSettings() {
        return this.http.get('/webhooks/settings');
    }
    /**
     * Update webhook settings
     */
    async updateSettings(params) {
        return this.http.post('/webhooks/settings', params);
    }
    /**
     * Get webhook logs
     */
    async getLogs() {
        return this.http.get('/webhooks/logs');
    }
    /**
     * Test webhook delivery
     */
    async test() {
        return this.http.post('/webhooks/test');
    }
    /**
     * Regenerate webhook secret
     */
    async regenerateSecret() {
        return this.http.post('/webhooks/regenerate-secret');
    }
    /**
     * Verify webhook signature
     */
    verify(payload, signature, secret) {
        const payloadString = typeof payload === 'string' ? payload : JSON.stringify(payload);
        const expectedSignature = crypto
            .createHmac('sha256', secret)
            .update(payloadString)
            .digest('hex');
        try {
            return crypto.timingSafeEqual(Buffer.from(signature), Buffer.from(expectedSignature));
        }
        catch {
            return false;
        }
    }
}
exports.Webhooks = Webhooks;
